docs/Roadmap.md

```diff
@@ -0,0 +1,140 @@
+currently provide similiar capabilities for monadic verification.  The
+standard library should provide monadic verification capabilities
+including pre and post-conditions of monadic code as well as
+loop-invariants.  Key challenges that will need to be addressed include:
+(1) ensuring a consistent experience when verifying code in different
+monads while allowing the pre and post-conditions to refer to monadic
+state; (2) use of symbolic execution to unfold large monadic
+computations while ensuring verification formulas are tractable; (3)
+addressing some foundational issues with references in `ST` and `IO`.
+
+### Primitive data types
+
+Leah has builtin support for a variety of primitive types including
+natural numbers, unsigned fixed-width integers (e.g., `UInt8`),
+double-precision floating point numbers, `String`, arrays (including
+scalar arrays), and other types.  This builtin support improves
+efficiency over the generic object representation and can facilitate
+better interoperability between Lean and C code.
+
+The standard library and Lean should additional types to improve the
+interoperability of Lean with other languages.  In particular, Lean
+should support fixed precision *signed* integers including `Int8`,
+`Int16`, `Int32` and `Int64`.  We also aim to provide aliases from C
+primitive types such as `int`, `long`, `ptrdiff_t` etc to facilitate
+Lean and C integration.
+
+### Data structures
+
+The standard library should provide common data structured needed in
+applications including ordered-maps, hash tables, arrays, lists, tries,
+sequences.  These need to be both performant and come with lemmas so
+they can be easily reasoned about in specifications.  Lean's runtime
+supports destructive updates when data structures are not shared, and
+the standard library needs to provide data structures suitable for both
+good shared and destructive use-cases.  Furthermore, Lean provides
+concurrency primitives, and the standard library needs to design data
+structures for concurrent accesses and updates.
+
+## Asynchronous IO
+
+Lean currently only provides IO functions that block the kernel thread
+while executing.  Unfortunately, this doesn't scale well in applications
+that involve large numbers of potential IO operations since each waiting
+system kernel call blocks a kernel thread while it is completed.  The
+standard library aims to provide a comprehensive asynchronous concurrent
+IO library for file access and monitoring, networking, IPC and general
+event-based programming that abstracts over the underlying kernel APIs
+that Lean supports.
+
+## Proof automation
+
+The standard library will tactics and other proof automation for the
+broader Lean community.  Our priorities in 2024 are:
+
+* Fast and efficient arithmetic constraint solving that can be
+  used independently and integrated into other decision procedures.
+
+* Support for an Isabelle Hammer-type tactic that can call a portfolio
```

Comment:
Link to a reference? This looks recent: https://isabelle.in.tum.de/dist/doc/sledgehammer.pdf
```suggestion
* Support for a tactic similar to Isabelle's Sledgehammer that can call a portfolio
```